### DISCUSSION TITLE
What determines the project nature in Theia?

### DISCUSSION THREAD

**Discussion by tsiyona** [eclipse-theia/theia#10215] [Q&A] [answered]
### Scenario #1:
I have a workspace which contains two java projects. Both projects have a pom.xml file.
When I open the workspace in Theia, for the first project, Theia adds '.project' and '.classpath' files, and marks the project with a **java** nature and a **maven** nature. The second project is marked only with a maven nature, and has only a '.project' file (no '.classpath' file).
Why is the difference?

### Scenario #2:
I have a workspace which contains two projects. The first project has a 'build.gradle' file. The second project has a pom.xml file.
When I open this workspace in Theia, the gradle project is added a '.project' file and is marked as a gradle project. The second project does not get a '.project' file at all.
When I query the workspace (with org.eclipse.core.resources.ResourcesPlugin API), only the first project is identified as  a project.

Can someone please explain to me what determines how Theia recognizes what constitutes a project (and of what nature) and what does not?

**Comment by msujew** [comment, marked as answer]
Hi @tsiyona and thank you for your question.

Java Support for Theia/vscode is usually contributed by the [Redhat Java extension](https://github.com/redhat-developer/vscode-java) and not by Theia itself. Theia is by design not aware of any language. All language support is contributed by extensions, just like vscode.

For the mentioned extension, there is already an issue regarding `.project` and `.classpath` files, see https://github.com/redhat-developer/vscode-java/issues/618. It seems like this is a side effect of using the `m2e` (maven integration for eclipse) library.

**Reply by tsiyona** [reply]
Thank you @msujew . I will continue my investigation in redhat-developer java extension group.
